src/lib/cmp.js

```diff
@@ -210,35 +210,43 @@ export default class Cmp {
 					if (config.gdprAppliesGlobally) {
 						// if no cookie, show tool
 						if (needsPublisherCookie || needsGlobalCookie) {
+							log.debug("GDPR applies globally and user needs a cookie");
 							cmp('showConsentTool');
 						}
 						// if cookie present and current, don't show tool
 						// if cookie present and old, show tool
 						else if (self.utils.checkReprompt(config.repromptOptions, vendorConsents, vendors)) {
+							log.debug("GDPR applies globally and user's cookie is old enough to show reprompt window");
 							cmp('showConsentTool');
 						}
 						else {
 							log.debug("rendering the CMP is not needed");
 						}
 					}
 					else {
-						// if not in EU, no cookie present, don't show tool
-						// if geolocation in EU, no cookie present, show tool
-						if (self.utils.checkIfGDPRApplies()) {
-							if (needsPublisherCookie || needsGlobalCookie) {
-								cmp('showConsentTool');
-							} else {
-								store.toggleFooterShowing(true);
-							}
-						}
-						// if cookie present and current, don't show tool
-						// if cookie present and old, show tool
-						else if (self.utils.checkReprompt(config.repromptOptions, vendorConsents, vendors)) {
-							cmp('showConsentTool');
-						}
-						else {
-							log.debug("rendering the CMP is not needed");
-						}
+						self.utils.checkIfGDPRApplies()
+							.then(applies => {
```

Comment:
Can you revert all the file? this would be changed in my PR
`checkIfGDPRApplies()` returns resolved promise and error is thrown - this would break demo page.